include/fuse_lowlevel.h

```diff
@@ -1954,6 +1968,36 @@ struct fuse_session *fuse_session_new(struct fuse_args *args,
 				      const struct fuse_lowlevel_ops *op,
 				      size_t op_size, void *userdata);
 
+/**
+ * Set a file descriptor for the session.
+ *
+ * This function can be used if you want to have a custom communication
+ * interface instead of using a mountpoint. In practice, this means that instead
+ * of calling fuse_session_mount() and fuse_session_unmount(), one could call
+ * fuse_session_custom_io() where fuse_session_mount() would have otherwise been
+ * called.
+ *
+ * In `io`, implementations for read and writev MUST be provided. Otherwise -1
```

Comment:
Why?

Reply:
If the user provides their own file descriptor, we can't guarantee that the default behavior of the io operations (the `read`, `writev`, `splice`) made in libfuse will function properly. Therefore, I think it's safest if we enforce the user to implement these io operations if they are the ones providing file descriptors

Reply:
Hmm. ok.

Reply:
I guess the explanation should be added as comment somewhere or at least in the commit message?

Reply:
Sure, I'llcan add it as a comment in the relevant if-block